src/main/scala/loamstream/drm/lsf/BacctAccountingClient.scala

```diff
@@ -0,0 +1,185 @@
+package loamstream.drm.lsf
+
+import java.time.Instant
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeFormatterBuilder
+import java.time.temporal.ChronoField
+
+import scala.util.Success
+import scala.util.Try
+import scala.util.matching.Regex
+
+import loamstream.conf.LsfConfig
+import loamstream.drm.AccountingClient
+import loamstream.drm.Queue
+import loamstream.model.execute.Resources.LsfResources
+import loamstream.model.quantities.CpuTime
+import loamstream.model.quantities.Memory
+import loamstream.util.Loggable
+import loamstream.util.Options
+import loamstream.util.RetryingCommandInvoker
+import loamstream.util.Tries
+
+/**
+ * @author clint
+ * Apr 18, 2019
+ */
+final class BacctAccountingClient(
+    bacctInvoker: RetryingCommandInvoker[String]) extends AccountingClient with Loggable {
+
+  import BacctAccountingClient._
+
+  override def getResourceUsage(jobId: String): Try[LsfResources] = {
+    getBacctOutputFor(jobId).flatMap(toResources)
+  }
+  
+  private def getBacctOutputFor(jobId: String): Try[Seq[String]] = bacctInvoker(jobId).map(_.stdout)
+    
+  private def toResources(bacctOutput: Seq[String]): Try[LsfResources] = {
+    val mungedBacctOutput = bacctOutput.map(_.trim)
+    
+    /*
+     * Documentation on bacct's output format:
+     * https://www.ibm.com/support/knowledgecenter/en/SSWRJV_10.1.0/lsf_command_ref/bacct.1.html
+     */
+    val node = mungedBacctOutput.collectFirst { case Regexes.node(n) => n }
+    
+    val queue = mungedBacctOutput.collectFirst { case Regexes.queue(q) => q }.map(Queue(_))
+    
+    val dataLineOpt = mungedBacctOutput.sliding(2).collectFirst {
+      case Seq(firstLine, nextLine) if isHeaderLine(firstLine) => nextLine.trim
+    }
+    
+    val dataLineAttempt = Options.toTry(dataLineOpt)(s"Couldn't find data line in bacct output: '$mungedBacctOutput'")
+    
+    dataLineAttempt.flatMap(parseDataLine(mungedBacctOutput, node, queue))
+  }
+}
+
+object BacctAccountingClient {
+  /**
+   * Make a QacctAccountingClient that will retrieve job metadata by running some executable, by default, `qacct`.
+   */
+  def useActualBinary(lsfConfig: LsfConfig, binaryName: String = "bacct"): BacctAccountingClient = {
+    new BacctAccountingClient(BacctInvoker.useActualBinary(lsfConfig.maxBacctRetries, binaryName))
+  }
+  
+  //"Data lines" look like this:
+  //CPU_T     WAIT     TURNAROUND   STATUS     HOG_FACTOR    MEM    SWAP
+  //0.02        0              0     exit         0.0000     0M      0M
+  def parseDataLine(
```

Comment:
Without knowing exactly how the data is generated or if the column order may change, have empty-string values, etc. it may be worth actually using a library to parse it (something like http://nrinaudo.github.io/kantan.csv/)? Not suggesting for this PR assuming what exists works, though.

Reply:
That's a good idea, since we have column headers.  It's unfortunate that `bacct`'s output isn't configurable to any great extent.  Many of the other `b` commands, like `bjobs` let you specify a format string where you lay out exactly what fields, with what padding, etc, that you want.  With `bacct`, you just get what you get.  If we used a lib to access things by column name, we'd at least get a more helpful error if the format changed and a column went away.